www/src/components/themes/ThemeSwitcher.vue

```diff
@@ -0,0 +1,78 @@
+<script setup lang="ts">
+import { type RemovableRef, useStorage } from '@vueuse/core';
+import { langId } from '@components/frontend/lang';
+import { defaultTheme, type Theme, themeKey } from '@components/themes/types';
+import IconButton from '@components/IconButton.vue';
+
+const translations = {
+    'en': {
+        'Change theme': 'Change theme',
+        'Themes': 'Themes',
+        'Modern': 'Modern',
+        'Classic (green)': 'Classic (green)',
+    },
+    'pl': {
+        'Change theme': 'Zmień motyw',
+        'Themes': 'Motywy',
+        'Modern': 'Nowoczesny',
+        'Classic (green)': 'Klasyczny (zielony)',
+    },
+};
+
+function translate(text: keyof (typeof translations)[LangId]): string {
+    return translations[langId][text];
+}
+
+const themeLabels: Record<Theme, string> = {
+    'modern': translate('Modern'),
+    'classic': translate('Classic (green)'),
+};
+
+const theme = useStorage(themeKey, defaultTheme) as RemovableRef<Theme>;
+
+function changeTheme(newTheme: Theme) {
+    theme.value = newTheme;
+    window.location.reload();
```

Comment:
The reload is unnecessary. It would be better to set `document.documentElement.dataset.bsTheme`.